### ISSUE TITLE
add gamma to CosineAnnealingWarmRestarts so max lr can decrease in cycles

### ISSUE THREAD

**Issue by Ir1d** [pytorch/pytorch#62640]
## 🚀 Feature

add gamma to CosineAnnealingWarmRestarts so max lr can decrease in cycles

## Motivation

In current [implementation](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingWarmRestarts.html), 
the max lr is the same in each cycle. It would be great if the max lr (`\eta_max` )can be set to decreasing.

## Pitch

add an `gamma` option to `CosineAnnealingWarmRestarts`, so the max learning rate decreases after each cycle

## Additional context

Similar implementation: https://github.com/katsura-jp/pytorch-cosine-annealing-with-warmup#args

I'll be glad to send a PR

**Comment by JohnNesbit** [comment]
I will start working on this project.
